service/history/history_engine.go

```diff
@@ -313,6 +315,10 @@ func NewEngineWithShardContext(
 		replicationTaskExecutorProvider,
 		dlqWriter,
 	)
+
+	historyEngImpl.versionMembershipCache = cache.New(10000, &cache.Options{
```

Comment:
I think the size should come from config too. In the past, non configurable cache sizes caused issues in produciton.

Also, can we have cache hit and mix metrics if the generic cache implementation does not provide it already? (so we can later reason about the size and ttl and adjust them)

Reply:
More importantly, the cache should be per node, not per shard, it seems we need to move it out of history_engine to somewhere else. I'm not sure where to but @prathyushpv @yycptt should know.